homeassistant/components/backup/config.py

```diff
@@ -0,0 +1,77 @@
+"""Provide persistent configuration for the backup integration."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Self, TypedDict
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.storage import Store
+from homeassistant.helpers.typing import UNDEFINED, UndefinedType
+
+from .const import DOMAIN
+
+STORAGE_KEY = DOMAIN
+STORAGE_VERSION = 1
+
+
+class StoredBackupConfig(TypedDict):
+    """Represent the stored backup config."""
+
+    last_automatic_backup: datetime | None
+    max_copies: int | None
+
+
+@dataclass(kw_only=True)
+class BackupConfigData:
+    """Represent loaded backup config data."""
+
+    last_automatic_backup: datetime | None = None
```

Comment:
This isn't really config, but we need to store this if we want to give consumers the last automatic backup time, regardless of the metadata of any backup files, and also if we want to be able to guard against too frequent backups eg due to DST time changes.